bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -1343,7 +1347,8 @@ private void checkStatus() {
             } else {
                 // Determine current status
                 jcheckType = "target jcheck";
-                checkablePullRequest.executeChecks(localHash, censusInstance, visitor, targetJCheckConf);
+                var issues = checkablePullRequest.executeChecks(localHash, censusInstance, visitor, targetJCheckConf);
+                tooFewReviewers = issues.stream().anyMatch(TooFewReviewersIssue.class::isInstance);
```

Comment:
I am not sure if it's a good idea to change the return value of `CheckablePullRequest.executeChecks`.

I think you can use `visitor.getChecks()` to determine if reviewers check failed or not.

```
        var tooFewReviewers = visitor.getChecks()
                .entrySet().stream()
                .anyMatch(entry -> entry.getKey().contains("Change must be properly reviewed") && entry.getValue().equals(false));
```

Reply:
> I am not sure if it's a good idea to change the return value of `CheckablePullRequest.executeChecks`.

Why?

> I think you can use `visitor.getChecks()` to determine if reviewers check failed or not.

Keys from `Map<String, Boolean> getChecks()` are to be displayed to an end user; they don't seem to be for programmatic consumption. I shudder when thinking about parsing these:

    "Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))"
    "Change must not contain extraneous whitespace"

Reply:
I was just thinking we should make the minimal necessary changes.  I admit that parsing text displayed to the end user isn't best practice, but we do it in many places of SKARA.

Reply:
It's just a suggestion, since you don't like it and I am ok with your pr, I will approve it now.

Reply:
Usually, making a minimally necessary change is good, and parsing end-user text is bad. In fact, I'd go to great lengths to avoid "stringly typed" code that operates on strings that are also subject to change.

Here, I initially thought I might be violating some encapsulation boundary by bringing a dependency. Maybe I still am. If nothing else, it's not the first class imported from that package:

     import org.openjdk.skara.jcheck.JCheckConfiguration;
    +import org.openjdk.skara.jcheck.TooFewReviewersIssue;

Reply:
I think it's ok to return the list of jcheck issues, but the variable name `issues` can be confusing as an `Issue` in the bots is usually some form of Jira Issue. I would recommend `jcheckIssues` to avoid confusion.

Reply:
Fair enough. I can see that "issue" is an overloaded term in the Skara domain:

  - org.openjdk.skara.issuetracker.Issue
  - org.openjdk.skara.jcheck.Issue

One word, two meanings. Does `jcheckIssues` sound better to you?

Reply:
Yes, I wrote just that in the previous comment from last week. :)

Reply:
> Yes, I wrote just that in the previous comment from last week. :)

D'oh. The PR has been open for quite a while, and I forgot that you asked that question. While I cannot guarantee that I didn't internalise and then forgot your question, it's good that we're now on the same page.